cfgov/jinja2/v1/_layouts/base.html

```diff
@@ -190,8 +190,15 @@
     {# Check and include template-level JavaScript. #}
     {% include '/js/routes/' + request.path.split('/')[1] + '/single.js' ignore missing %}
 
+    {# TODO: Remove conditional when this is fully moved to wagtail. #}
     {# Check and include page-level JavaScript. #}
-    {% include '/js/routes/' + request.path[1:] + 'index.js' ignore missing %}
+    {% if page %}
+        {% for js in page.media %}
```

Comment:
I personally use [django-webpack-loader](https://github.com/owais/) to insert React elements in some of my wagtail blocks and it's awesome : I can pass data from the django view to the React component (optionally set by the Wagtail Editor) just like this:
```
{% load render_bundle from webpack_loader %}
{% load downloads_tags  %}

{% get_products as products %}

<div class="container">
    {% render_bundle 'Downloads' %}

    <div class="col-sm-7">
        <div id="downloads"></div>
        <script type="text/javascript">
            Downloads.render({downloadEndpoint: '/download/', products:{{ products|safe }}}, 'downloads')
        </script>
    </div>
</div>
```

And during development, I have [React Hot Loader](http://gaearon.github.io/react-hot-loader/) working.